openfisca_core/holders.py

```diff
@@ -211,8 +211,7 @@ def _set(self, period, value, extra_params = None):
         if self.variable.definition_period != ETERNITY:
             if period is None:
                 raise ValueError('A period must be specified to set values, except for variables with ETERNITY as as period_definition.')
-            if ((self.variable.definition_period == MONTH and period.unit != periods.MONTH) or
-               (self.variable.definition_period == YEAR and period.unit != periods.YEAR)):
+            if self._is_not_month(period) or self._is_not_year(period):
```

Comment:
This PR is ostensibly for a minor upgrade of our linter, I'd prefer if the changes we make as a result of that were strictly the minimum necessary to pass the new rules: here for instance, moving the "or" from the end of one line to the beginning of the next.

Refactoring the code in `holders.py` is a worthwhile exercises and I'm very much in favor of allocating some time to that, to be done in pairs or mobs so we can get better aligned on our sense of good code design. Let's do that, with greater focus, as a task separate from upgrading linters.

Reply:
On reflection, it seemed best to add W504 to ignored rules, as we already have W503 in there for the same reason: "house rule is to break after binary operators".